### PR TITLE
Feature/ahoy events setup

### PR DESCRIPTION
**Pull request by abdokin**

### CODE REVIEW

File: app/controllers/candidate/assessment_participations_controller.rb [hunk 1]
```diff
@@ -16,11 +16,21 @@ def invitation
   end
 
   def overview
+    ahoy.track "Candidate joined", {
+      assessment: @assessment.title,
+      business: @assessment.business.company
+    }
```
Comment (itsAndL) [inline app/controllers/candidate/assessment_participations_controller.rb:22]: Remember the principle "Skinny model and skinny controller", try to put those within that service

File: app/controllers/candidate/assessment_participations_controller.rb [hunk 2]
```diff
@@ -16,11 +16,21 @@ def invitation
...
     @participation_service.invitataion_clicked
   end
 
   def setup
     @next_url = @participation_service.start
+    
+    ahoy.track "Assessment started", {
+      assessment: @assessment.title,
+      business: @assessment.business.company
+    }
```
Comment (itsAndL) [inline app/controllers/candidate/assessment_participations_controller.rb:33]: Remember the principle "Skinny model and skinny controller", try to put those within that service

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -3,4 +3,17 @@ class ApplicationController < ActionController::Base
...
+
+  private
+  
+  def track_action
+      ahoy.track "#{controller_name}##{action_name}", request.path_parameters
+  end
+
+  def set_ahoy_visitor
+    ahoy.authenticate(current_user) if current_user
+  end
```
Comment (itsAndL) [inline app/controllers/application_controller.rb]: Put those in a module

File: db/migrate/20241104092048_create_ahoy_visits_and_events.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
+class CreateAhoyVisitsAndEvents < ActiveRecord::Migration[7.2]
```
Comment (itsAndL) [inline db/migrate/20241104092048_create_ahoy_visits_and_events.rb:1]: Run this migration, the schema hasn't yet updated

File: app/models/assessment.rb [hunk 5]
```diff
@@ -3,6 +3,8 @@ class Assessment < ApplicationRecord
   include PgSearch::Model
   include LanguageEnum
 
+  after_create :track_action
```
Comment (itsAndL) [inline app/models/assessment.rb:6]: you can rename "track_action" to "track_event"

**Comment by abdokin** [comment]
[Track events with ahoy](https://trello.com/c/UgM5lzUU/86-track-events-with-ahoy)

**Review by itsAndL — APPROVED** [review]

**Pull request merged**